### PR TITLE
Products listing

### PR DESCRIPTION
**Pull request by ravikumar1002**
* Products listing Page
 * Filter 
     - Sort by
     - Ratings
     - Range
     - Category

https://user-images.githubusercontent.com/86847625/161383869-de67cfe2-fc43-44b8-b7b0-77550672f615.mp4

### CODE REVIEW

File: src/reducer/filter/filter-reducer.jsx [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+            })
+            .filter((rating) => {
+                return ratingValue === null ? true : rating.ratings >= ratingValue;
+            })
+            .filter((category) => {
+                return categoryValue.length > 0
+                    ? categoryValue.includes(category.categoryName)
+                    : true;
+            });
+    }
```
Comment (thegeorgenikhil) [inline src/reducer/filter/filter-reducer.jsx:25]: This is a really good implementation.

**Review by thegeorgenikhil — APPROVED** [review]
Really good function implementations for sorting and filtering.
Clean and easily understandable code.
Would be better if there were a deployed link to test out the functionalities

**Review by manish591 — APPROVED** [review]
Code is looking good to merge, naming is good and code is properly formatted

**Comment by ravikumar1002** [comment]
> Really good function implementations for sorting and filtering. Clean and easily understandable code. Would be better if there were a deployed link to test out the functionalities

yes Nikhil, from next project I will surely share active link

File: src/reducer/filter/filter-reducer.jsx (hunk 1 above)
Comment (ravikumar1002) [reply src/reducer/filter/filter-reducer.jsx:25]: thank u Nikhil

**Pull request merged**